src/tests/ftest/util/dfuse_utils.py

```diff
@@ -137,18 +137,25 @@ def create_mount_point(self):
 
         _, missing_nodes = check_file_exists(
             self.hosts, self.mount_dir.value, directory=True)
-        if len(missing_nodes):
-
-            cmd = "mkdir -p {}".format(self.mount_dir.value)
-            ret_code = pcmd(missing_nodes, cmd, timeout=30)
-            if len(ret_code) > 1 or 0 not in ret_code:
-                error_hosts = NodeSet(
-                    ",".join(
-                        [str(node_set) for code, node_set in ret_code.items()
-                         if code != 0]))
-                raise CommandFailure(
-                    "Error creating the {} dfuse mount point on the following "
-                    "hosts: {}".format(self.mount_dir.value, error_hosts))
+        if missing_nodes:
+            # Remove any filenames matching the directory name to avoid errors
+            command_list = [
+                "ls -al {}".format(self.mount_dir.value),
+                "rm -fr {}".format(self.mount_dir.value),
```

Comment:
Just as a heads up, if this directory exists at this point it's likely because a previous test has errored catastrophically and not performed dfuse shutdown, if that's the case the kernel may still register the mount but there will be no dfuse process so this 'rm -rf' command would block forever.

Reply:
Is there anything reasonably better we can do in this situation? Hopefully the updated `stop()` ran correctly and handled things correctly, but we are not 100% immune from it occurring.  At least if what you described occurs we should hit the test timeout and the signature should indicate what went wrong, or at least what to look at next (a previous test), no?

Reply:
I did see a case of this in the https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3318/20/artifact/Functional/job-2020-09-04T02.46-65ffa89-io-daos_vol/job.log, where test 8 timed out and therefore was not able to run all of its tearDown() steps.  This basically left everything running (dfuse, mpirun, agents, servers, etc.) and caused test 9 to fail to start - specifically with the creation of the dfuse mount point directory.  Most of test 8's mess was cleaned up by test 9's tearDown - as they use the same servers - but dfuse was not cleaned up because in test 9 the Dfuse.running_hosts was never set - which is expected as Dfuse.run() was not called due to the failed Dfuse.create_mount_point().

To resolve as much of this as is possible in the next test, which is attempting to use the same dfuse mount point, I've added a 'fusermount{,3} -uz ...' command in Dfuse.create_mount_point().

Reply:
We did have some discussion at the time about the possibility if having tearDown() fail so badly that it prevented future tests from even running, but apparently it does not have this power.  Next best (and alongside that) is to have this code detect the problem and immediately fail, rather than attempting to cleanup with the associated risk of deadlock.

I don't think there is a 100% reliable way of stopping fuse after abnormal termination, and chasing it is a moving target so my preferred is to put in a sensible amount of effort in stop() to work in common failure cases and have the start() fail immediately and loudly, rather than in obscure ways or deadlocking.